src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -299,19 +309,26 @@ def get_callbacks_watchlist(train_cfg, train_dmatrix, val_dmatrix, model_dir, ch
         logging.info("Resuming from iteration %s", iteration)
 
     callbacks = []
-    callbacks.append(checkpointing.print_checkpointed_evaluation(start_iteration=iteration))
+    callbacks.append(xgb.callback.EvaluationMonitor())
     if checkpoint_dir:
-        save_checkpoint = checkpointing.save_checkpoint(checkpoint_dir, start_iteration=iteration)
-        callbacks.append(save_checkpoint)
+        callbacks.append(xgb.callback.TrainingCheckPoint(checkpoint_dir, iterations=iteration))
 
     # Parse arguments for intermediate model callback
     save_model_on_termination = train_cfg.pop('save_model_on_termination', "false")
     if save_model_on_termination == "true":
         model_name = f"{MODEL_NAME}-{fold}" if fold is not None else MODEL_NAME
-        save_intermediate_model = checkpointing.save_intermediate_model(model_dir, model_name)
-        callbacks.append(save_intermediate_model)
+        callbacks.append(checkpointing.SaveIntermediateModelCallBack(model_dir, model_name))
         add_sigterm_handler(model_dir, is_master)
 
+    if early_stopping_data and early_stopping_metric and early_stopping_rounds:
```

Comment:
If val_dmatrix is None, is it better to skip early stopping?

Reply:
It is very dangerous to early stop on the training data. I believe that MLP has the same design. Without validation data, MLP won't do early stopping.

Reply:
I agree. Now if early_stopping_data == 'train', the early stopping will still be used, no?

Reply:
yes. But `early_stopping_data` is not a customer parameter (it is even not a string). We set it to `val_matrix` or `None`

Reply:
fixed